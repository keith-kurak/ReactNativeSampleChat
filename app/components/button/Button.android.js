import React, { Component } from 'react';
import { Text, View,  TouchableHighlight } from 'react-native';

import Styles from './styles';

export default class Button extends Component {
  constructor(props) {
    super(props);
  }

  _onPressButton() {
    console.log("You tapped the button!");
    if(this.props.onButtonPress) {
      this.props.onButtonPress();
    }
  }

  render() {

    // using this to demonstrate platform-specific components. Not meant to really do this with a button.
    return (
        <View>
          <Text>-Android-</Text>
          <View style={[Styles.circleArea, Styles.buttonStyle, Styles.smallSpace]}>
            <TouchableHighlight style={[Styles.circleArea, Styles.container]} onPress={() => this._onPressButton()}>
              <Text>{this.props.text}</Text>
            </TouchableHighlight>
          </View>
        </View>
      );
  }
}