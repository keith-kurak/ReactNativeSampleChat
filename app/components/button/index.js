//non-default export (better for expansion/ autocomplete?)
export {default as Button} from './Button';

//default export (some people say don't do this)
//import Button from './button.js';
//export default Button;