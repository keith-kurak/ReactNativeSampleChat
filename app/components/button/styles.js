import { StyleSheet } from 'react-native';

export default StyleSheet.create({

  smallSpace: {
    margin: 5
  },

  container: {
    justifyContent: 'center',
    alignItems: 'center'
  },

  circleArea: {
    width: 60,
    height: 60,
    borderRadius: 60/2
  },

  buttonStyle: {
    backgroundColor: 'white',
    borderColor: 'black',
    borderWidth: .5
  },

  buttonText: {
    fontSize: 30,
    color: 'black',
    fontWeight: 'bold'
  }
});