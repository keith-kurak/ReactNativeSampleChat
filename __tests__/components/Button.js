import 'react-native';
import React from 'react';
import {Button} from '../../app/components/button';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('render with text set matches snapshot', () => {
  const tree = renderer.create(
    <Button text={'some text'} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});