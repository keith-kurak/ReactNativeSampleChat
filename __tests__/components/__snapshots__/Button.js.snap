// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with text set matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 30,
        "height": 60,
        "width": 60,
      },
      Object {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderWidth": 0.5,
      },
      Object {
        "margin": 5,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Array [
          Object {
            "borderRadius": 30,
            "height": 60,
            "width": 60,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ],
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      some text
    </Text>
  </View>
</View>
`;
